climada/engine/uncertainty/unc_impact.py

```diff
@@ -0,0 +1,309 @@
+        ValueError:
+            If no sampling parameters defined, the distribution cannot
+            be computed.
+
+        """
+
+        if self.sample.empty:
+            raise ValueError("No sample was found. Please create one first"
+                             "using UncImpact.make_sample(N)")
+
+        if rp is None:
+            rp=[5, 10, 20, 50, 100, 250]
+
+        self.rp = rp
+        self.calc_eai_exp = calc_eai_exp
+        self.calc_at_event = calc_at_event
+        
+        start = time.time()
+        one_sample = self.sample.iloc[0:1].iterrows()
+        imp_metrics = map(self._map_impact_calc, one_sample)
+        [aai_agg_list, freq_curve_list,
+         eai_exp_list, at_event_list, tot_value_list] = list(zip(*imp_metrics))
+        elapsed_time = (time.time() - start) 
+        est_com_time = self.est_comp_time(elapsed_time, pool)
+        LOGGER.info(f"\n\nEstimated computation time: {est_com_time}s\n")
+        
+        #Compute impact distributions
+        if pool:
+            LOGGER.info('Using %s CPUs.', pool.ncpus)
+            chunksize = min(self.n_runs // pool.ncpus, 100)
+            imp_metrics = pool.map(self._map_impact_calc,
+                                           self.samples.iterrows(),
+                                           chunsize = chunksize)
+
+        else:
+            imp_metrics = map(self._map_impact_calc, self.sample.iterrows())
+        
+        #Perform the actual computation
+        with log_level(level='ERROR', name_prefix='climada'):
+            [aai_agg_list, freq_curve_list,
+             eai_exp_list, at_event_list,
+             tot_value_list] = list(zip(*imp_metrics))
+            
+
+        # Assign computed impact distribution data to self
+        self.metrics['aai_agg']  = pd.DataFrame(aai_agg_list,
+                                                columns = ['aai_agg'])
+
+        self.metrics['freq_curve'] = pd.DataFrame(freq_curve_list,
+                                    columns=['rp' + str(n) for n in rp])
+        self.metrics['eai_exp'] =  pd.DataFrame(eai_exp_list)
+        self.metrics['at_event'] = pd.DataFrame(at_event_list)
+        self.metrics['tot_value'] = pd.DataFrame(tot_value_list,
+                                                 columns = ['tot_value'])
+        self.check()
+        
+        return None
+
+    def _map_impact_calc(self, sample_iterrows):
+        """
+        Map to compute impact for all parameter samples in parrallel
+
+        Parameters
+        ----------
+        sample_iterrows : pd.DataFrame.iterrows()
+            Generator of the parameter samples
+
+        Returns
+        -------
+         : list
+            impact metrics list for all samples containing aai_agg, rp_curve,
+            eai_exp (np.array([]) if self.calc_eai_exp=False) and at_event
+            (np.array([]) if self.calc_at_event=False).
+
+        """
+
+        # [1] only the rows of the dataframe passed by pd.DataFrame.iterrows()
```

Comment:
the [1] hence shouldnt be necessary when using the apply instead of the iterrows() / map() syntax

Reply:
I do not see how you could use the apply method. Could you explain how you would do it? Then one could use `dask` for parallelization.

Reply:
`impact_metrics = self.samples_df.apply(_map_impact_calc)` ?